server/src/main/java/org/elasticsearch/http/HttpRequest.java

```diff
@@ -68,6 +68,10 @@
      */
     HttpResponse createResponse(RestStatus status, BytesReference content);
 
+    boolean hasInboundException();
```

Comment:
This method seems to only be used in tests. Just make `getInboundException` `@Nullable` and use that to keep things tidy?